nemo/collections/asr/parts/submodules/multitask_greedy_decoding.py

```diff
@@ -205,7 +205,7 @@ def forward(
 
         return (packed_result,)
 
-    def format_hypotheses(self, packed_result: List[Hypothesis], decoder_input_ids: torch.Tensor | None) -> None:
+    def format_hypotheses(self, packed_result: List[Hypothesis], decoder_input_ids: Union[torch.Tensor, None]) -> None:
```

Comment:
I think there're many other files that use `|` instead of `Union`, any thoughts on whether we should update this? I would prefer `Union` for backward compatibility with python 3.9, but seems nemo already fixes to python to 3.10

Reply:
I think all the others do not have this [forward reference issue](https://docs.python.org/3/library/stdtypes.html#union-type:~:text=Note%20The%20%7C,%7C%20Foo%22.)? So we can leave all the others for now?

Reply:
Oh you're right, there are a bunch more, you can see them if you go [here](https://readthedocs.com/projects/nvidia-nemo/builds/2416189/) and click on the box that says `python -m sphinx -T -b html -d _build/doctrees -D language=en . $READTHEDOCS_OUTPUT/html`.

Then if you ctrl-f for "typeerror: unsupported operand type(s) for |:", there are 56 results.
Weirdly, I do not see all these errors when I build the docs locally. I fixed all the ones that I saw locally, but there are many more.

Yes, we want to fix all these in the next week or 2. If you have time that would be great, or I can also do it no problem.
Docs team wants to get rid of all the warnings, so we can set the Docs checks to error if there are any warnings, so we can be alerted when the docs don't look right.

Reply:
Actually, I will have a go at these now

Reply:
NVM, fortunately there is no discrepancy between errors when I build docs locally and via CI. I got this impression because when I merged this branch to main, there was a file containing this type hinting error again (came from this PR https://github.com/NVIDIA/NeMo/pull/10086 cc @pzelasko ).

There were a lot of errors shown in the log, but they were all due to this 1 line, I have now fixed it.

Latest docs CI log has no more "typeerror: unsupported operand type(s) for |:" errors: https://readthedocs.com/projects/nvidia-nemo/builds/2416353/.